### ISSUE TITLE
redis-cli --latency doesn't capture SIGKILL when CMD

### ISSUE THREAD

**Issue by bogdanbarna** [docker-library/redis#251]
Hello, 

I noticed that if I run the following commands:

```
# redis 2.8.6
docker run -i --rm --network container:<my-redis-server-container> redis:2.8.6 redis-cli --latency -h localhost

# redis 6.0.5
docker run -i --rm --network container:<my-redis-server-container> redis:6.0.5-buster redis-cli --latency -h localhost
```

CTRL+C is not being captured.

However, if I run the `--latency` switch from inside the container, the signal is captured as expected:

```
docker run -i --rm --network container:<my-redis-server-container> redis:2.8.6 bash
redis-cli --latency -h localhost
```

Any idea on why this is happening?

**Comment by wglambert** [comment]
You can use `ctrl+pq` to detach from the container when other signals aren't working, you should also use `-t` with `-i` for a pseudo terminal. The issue is that `redis-cli` becomes PID 1 in that container and it isn't designed to handle those signals by itself, but `redis-server` is able to, so on a normal startup `redis-server` is PID 1 and handles signals appropriately.

Other alternatives are to start your second container with `--init` so that `redis-cli` isn't PID 1, also you could `docker exec -it <my-redis-server-container> redis-cli --latency -h localhost`

**Comment by Rundfunk** [comment]
I encountered the same issue while running the `MONITOR` command. Ctrl+c would not stop the monitoring, only printing `^C` in the console. As I use Docker Compose to manage my containers, I was able to a new container that connects to the container running the Redis server and execute the needed command.  

```
redis_monitor:
  image: library/redis:6.0.5-alpine3.12
  init: true
  depends_on:
    - redis
  entrypoint: ["redis-cli", "-h", "redis", "MONITOR"]
```
Note: in this example the container running Redis is called `redis`, adapt the values in `depends_on` and `entrypoint` to your own situation.

The fix is in the `init: true` line, which [runs a separate init binary inside the container and forwards any signals](https://docs.docker.com/compose/compose-file/#init), including SIGINT.

Now, running the container is as simple as:

```
docker-compose run --rm redis_monitor
```
